tekton/resources/ci/github-template.yaml

```diff
@@ -77,6 +78,19 @@ spec:
 
             # we may need to URLEncode the check name
             status_path = "$(resources.outputs.pr.path)/status/$(tt.params.checkName).json"
+            with open(status_path, "r") as status:
+              old_status = json.loads(status)
+
+            # Detect race condition. It may be that the "start" event is processed after
+            # the "complete" one. This does not eliminate the possibility of race as the two
+            # update jobs may still run in parallel, but it reduces the chances considerably.
+            if (status_body['State'] in ['success', 'failure', 'error'] and
```

Comment:
Doesn't this prevent from going from `success` or `failure` to `pending` (in case of an update on the PR) ?

Reply:
It shouldn't because the target URL includes the PR name, which is unique per run, so it will be different after a PR sync (or event after a `/test`

Reply:
oh nice :+1: